hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -713,6 +776,26 @@ discard :: Monad m => PropertyT m a
 discard =
   PropertyT $ lift Gen.discard
 
+-- | Add a classifier to the test if the predicate is true
+--
+-- @
+--    prop_with_classifier :: Property
+--    prop_with_classifier = property $ do
+--      xs <- forAll $ Gen.list (Range.linear 0 100) Gen.alpha
+--      for_ xs $ \x -> do
+--        classify (x == 0) "newborns"
+--        classify (x > 0 && x < 13) "children"
+--        classify (x > 12 && x < 20) "teens"
+--        success
+-- @
+classify :: (MonadTest m, HasCallStack) => Bool -> String -> m ()
+classify = cover 0
+
+cover :: (MonadTest m, HasCallStack) => Double -> Bool -> String -> m ()
```

Comment:
I would like to have a comment for this as it will be visible in the haddock.

@mankyKitty maybe we should mention this in the style guide, public facing functions should have a comment. Internal ones I think it's developer's discretion.

Reply:
Sounds good